devel/libecl/src/smspec_node.c

```diff
@@ -736,6 +736,38 @@ smspec_node_type * smspec_node_alloc_lgr( ecl_smspec_var_type var_type ,
   }
 }
 
+smspec_node_type* smspec_node_copy( const smspec_node_type* node ) {
+    if( !node ) return NULL;
+
+    smspec_node_type* copy = calloc( 1, sizeof( smspec_node_type ) );
+    UTIL_TYPE_ID_INIT(copy, node->__type_id)
```

Comment:
Have tried to not access the `__type_id` variable explicitly; so would prefer initialization based on the #define symbol `SMPEC_TYPE_ID`